### PR TITLE
source: csv: Enable setting repo src_url using CSVSourceConfig

### PR DESCRIPTION
**Pull request by sudharsana-kjl**
This is a fix for issue #48. Here I've included CSVSourceConfig separately so that FileSourceConfig remains undisturbed with respect to addition of ```key``` parameter.

### CODE REVIEW

File: dffml/source/csv.py [hunk 1]
```diff
@@ -21,6 +30,20 @@ class CSVSource(FileSource, MemorySource):
     # Headers we've added to track data other than feature data for a repo
     CSV_HEADERS = ["prediction", "confidence", "classification"]
 
+    @classmethod
+    def args(cls, args, *above) -> Dict[str, Arg]:
+        cls.config_set(args, above, "key", Arg(type=str, default=None))
```
Comment (pdxjohnny) [inline dffml/source/csv.py]: Don't forget to `config_set` all the variables

File: tests/source/test_csv.py [hunk 2]
```diff
@@ -3,17 +3,66 @@
 import unittest
 
 from dffml.source.file import FileSourceConfig
-from dffml.source.csv import CSVSource
+from dffml.source.csv import CSVSource, CSVSourceConfig
 from dffml.util.testing.source import FileSourceTest
 from dffml.util.asynctestcase import AsyncTestCase
+from dffml.repo import Repo
+from dffml.util.cli.arg import parse_unknown
+import tempfile
```
Comment (pdxjohnny) [inline tests/source/test_csv.py]: Same as the comment on `typing`

File: tests/source/test_csv.py [hunk 3]
```diff
@@ -3,17 +3,66 @@
...
+                "--source-file-readonly",
+            )
+        )
+        self.assertEqual(config.filename, "feedface")
+        self.assertEqual(config.label, "default-label")
+        self.assertEqual(config.key, "SourceURLColumn")
+        self.assertTrue(config.readonly)
+
+    def config(self, filename, label="unlabeled", key=None, readonly=False):
+        return FileSourceConfig(
```
Comment (pdxjohnny) [inline tests/source/test_csv.py]: CSVSourceConfig?

File: dffml/source/csv.py [hunk 4]
```diff
@@ -8,11 +8,20 @@
 
 from ..repo import Repo
 from .memory import MemorySource
-from .file import FileSource
+from .file import FileSource, FileSourceConfig
+from typing import NamedTuple, Dict
```
Comment (pdxjohnny) [inline dffml/source/csv.py]: Let's move this `typing` import out of the `dffml` imports and up to where we import from the rest of the standard library

File: dffml/source/csv.py [hunk 5]
```diff
@@ -45,6 +78,11 @@ class CSVSource(FileSource, MemorySource):
                         repo_data["features"][key] = ast.literal_eval(value)
                     except (SyntaxError, ValueError):
                         repo_data["features"][key] = value
+                if self.config.key is not None and self.config.key == key:
+                    src_url = value
+                if self.config.key is None:
+                    src_url = str(i)
+            i += 1
```
Comment (pdxjohnny) [inline dffml/source/csv.py]: i += 1 needs to be in the loop

File: tests/source/test_csv.py (hunk 3 above)
Comment (sudharsana-kjl) [reply tests/source/test_csv.py]: Sorry :sweat_smile:

File: dffml/source/csv.py (hunk 5 above)
Comment (sudharsana-kjl) [reply dffml/source/csv.py]: Oh, i think it has to outside the loop. Because inside the loop we are looking at only one Repo's key, value pairs.

Comment (sudharsana-kjl) [reply dffml/source/csv.py]: i+=1 still happens inside the main loop which runs for all the repos. It gets incremented only after we assign it to a particular repo. If i move it inside the inner loop as well, then it has to be incremented for each key, value pair from the data in a row. So this should be added outside the inner loop but within the main loop.

Comment (pdxjohnny) [reply dffml/source/csv.py]: Ah yes my bad!

File: tests/source/test_csv.py [hunk 6]
```diff
@@ -1,19 +1,68 @@
...
+                "SourceURLColumn",
+                "--source-file-readonly",
+            )
+        )
+        self.assertEqual(config.filename, "feedface")
+        self.assertEqual(config.label, "default-label")
+        self.assertEqual(config.key, "SourceURLColumn")
+        self.assertTrue(config.readonly)
+
+    def config(self, filename, label="unlabeled", key=None, readonly=False):
```
Comment (pdxjohnny) [inline tests/source/test_csv.py]: Looks like this `config()`  method is never called, can you please remove it? https://codecov.io/gh/intel/dffml/pull/102/src/tests/source/test_csv.py#L51

Comment (sudharsana-kjl) [reply tests/source/test_csv.py]: Done :+1:

**Pull request merged**